app/controllers/sites/scope_filters_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Sites
+	class ScopeFiltersController < ApplicationController
+		include Dcv::RestrictableController
+		include Cul::Omniauth::AuthorizingController
+		before_filter :load_subsite
+		before_filter :authorize_site_update
+		layout 'sites'
+
+		def load_subsite
+			@subsite ||= begin
+				site_slug = params[:site_slug]
+				site_slug = "restricted/#{site_slug}" if restricted?
+				Site.find_by(slug: site_slug)
+			end
+		end
+
+		def update
+			scope_filters_attributes = scope_filter_params['scope_filters_attributes']&.values || []
+			if scope_filters_attributes # must permit empty collection to allow deletions
+				begin
+					@subsite.scope_filters.each do |scope_filter|
+						if scope_filters_attributes.first
+							scope_filter.update_attributes(scope_filters_attributes.shift)
+						else
+							scope_filter.delete
+						end
+					end
```

Comment:
To confirm, this block is deleting scope filters that have been provided without a value?  If so, might be clearer to make it a separate method with naming that indicates that.

Reply:
Or I guess I should confirm that it's doing two things: updating the filter if present or deleting it if the value is blank.

Reply:
Alternately, a couple of extra comments inside of the `if` and `else` may clear things up.

Reply:
Yes, this is similar to other spots where a list of values is provided - if the input list is longer than the persisted list, there are additions; if shorter, deletions. Inasmuch as it is possible, it updates existing filters instead of wholesale delete all/add all.

Reply:
I can definitely add some comments.

Reply:
Okay, I think I figured out what was throwing me off here.  Is it always safe to assume that `@subsite.scope_filters` will always be a list of filters in the same order as what's being submitted in the scope_filters_attributes param?

Reply:
Yes - that's the assumption.

Reply:
It seems like the kind of thing that someone might break unintentionally in the future if they change the view and don't realize that the controller functionality is order dependent.

I'd recommend either:
- Destroying all site filters and re-creating them (safe, but more expensive)
or
- Expecting ScopeFilter ids to be sent with each scope filter attribute (which is the more traditional way of doing nested model param updates in rails, but is more annoying to do when you're building repeatable form elements dynamically)

Reply:
I don't understand how the former is different?

Reply:
Well, my understanding isn't the issue - if you think it would be clearer, I'll make the destroy all/add all change.

Reply:
(making that change will also mean that the filters are destroyed and added whether their attributes change or not, but as you point out this is probably a rare operation)

Reply:
Ah, got it now.  The `scope_filter.update_attributes(scope_filters_attributes.shift)` is updating both the `filter_type` and `value` attributes.  The code makes it seem like you're doing a patch change when in effect, the changes that you're making are equivalent creating a new ScopeFilter (since a ScopeFilter is basically just a filter_type+value).  I didn't originally get the intent, but you're trying to save one extra SQL query here (update vs. delete+create).

I think the delete+create would be slightly clearer (since performance doesn't matter in this case), but maybe it would be good to get a second opinion on this in case I'm overthinking it.

Reply:
If this were in a controller that might receive more updates with this data unaffected, I'd feel differently - but because this controller *only* handles updates to scope filters (they're not in params elsewhere), I think ease of reading is the right priority.